src/validator.directive.js

```diff
@@ -319,6 +319,9 @@
          */
         attrs.$observe('validator', function(value) {
           validation = value.split(',');
+          // validation has been updated => make rule fire again
+          var theValue = useViewValue ? ctrl.$viewValue : ctrl.$modelValue;
+          checkValidation(scope, element, attrs, ctrl, validation, theValue);
```

Comment:
The tests fail because the `checkValidation` will be called in the initialization phase, it should trigger only there is a change on `validator`.